website_product_comment_purchased/models/mail.py

```diff
@@ -22,41 +22,41 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp import models, fields, api
+from openerp.osv import osv, fields
 
 
-class MailMessage(models.Model):
+class MailMessage(osv.Model):
     _inherit = 'mail.message'
 
-    @api.multi
-    def _comment_bought(self):
-        sale_report_obj = self.env['sale.report']
-        sale_select = sale_report_obj._select()
-        sale_from = sale_report_obj._from()
-        sale_group_by = sale_report_obj._group_by()
-        for mail in self:
-            where = """
+    def _comment_bought(self, cr, uid, ids, name, arg, context=None):
+        res = {}.fromkeys(ids, False)
+        sale_report_obj = self.pool.get('sale.report')
+        mail_cache = self.browse(cr, uid, ids, context)[0]
+        where = """
 
-            WHERE l.product_id IS NOT NULL AND
-                        t.id={res_id} AND s.partner_id={partner_id}
+        WHERE l.product_id IS NOT NULL AND
+                   t.id={res_id} AND s.partner_id={partner_id}
 
-                """.format(res_id=mail.res_id,
-                           partner_id=mail.author_id.id)
-            execute = """
-                {select}
-                FROM ( {_from} )
-                {where}
-                {group}
-                LIMIT 1
-                """.format(
-                select=sale_select,
-                _from=sale_from,
-                where=where,
-                group=sale_group_by)
-            self._cr.execute(execute)
-            if self._cr.fetchone():
-                mail.comment_bought = True
+            """.format(res_id=mail_cache.res_id,
+                       partner_id=mail_cache.author_id.id)
+        execute = """
+            {select}
+            FROM ( {_from} )
+            {where}
+            {group}
+            LIMIT 1
+            """.format(
+            select=sale_report_obj._select(),
```

Comment:
I gues set that values at the top of method must to avoid recalculation in a for loop, but see the next:
![image](https://cloud.githubusercontent.com/assets/10233975/13115080/02fc5d76-d55c-11e5-982d-1a3e30843df9.png)